### ISSUE TITLE
torch.multinomial 而不是 argmax()

### ISSUE THREAD

**Issue by Peterande** [bytedance/1d-tokenizer#51]
作者您好，在RAR sample的过程中，我发现将`sampled = torch.multinomial(probs, num_samples=1)`改成`torch.argmax(probs, dim=-1)`，生成的结果完全崩坏、没有意义。这是什么原因导致的呢？ARLoss中算token是否正确不就是用的`torch.argmax(shift_logits, dim=1)`吗？

**Comment by cornettoyu** [comment]
Hi,

Thanks for sharing the findings. Unfortunately I do not have experience on using argmax (greedy search) for image generation, so I may not be able to share any insights on this
